kie-dmn-api/src/main/java/org/kie/dmn/api/core/DMNMessage.java

```diff
@@ -17,20 +17,62 @@
 package org.kie.dmn.api.core;
 
 import org.kie.dmn.api.feel.runtime.events.FEELEvent;
+import org.kie.dmn.feel.model.v1_1.DMNElement;
 
+/**
+ * A general message interface for all DMN related messages
+ * raised during compilation and execution.
+ */
 public interface DMNMessage {
 
     enum Severity {
         TRACE, INFO, WARN, ERROR;
     }
 
+    /**
+     * Returns the severity of the message. Either TRACE, INFO, WARN or ERROR
+     *
+     * @return
+     */
     Severity getSeverity();
 
+    /**
+     * Returns a human readable text with the explanation of the event that
+     * raised the message.
+     *
+     * @return
+     */
     String getMessage();
 
+    /**
+     * Returns the ID of the model element to which this message relates to
+     * or null if this message does not refer to a specific model element.
+     *
+     * @return
+     */
     String getSourceId();
 
+    /**
+     * Returns the actual model element reference to which this message relates to
+     * or null if this message does not refer to a specific model element.
+     *
+     * @return
+     */
+    DMNElement getSourceReference();
```

Comment:
@tarilabs adding this attribute to the message might break your kie-server marshalling. We need to check. The reason I am adding it here is to facilitate tooling integration.